File: core/model/src/main/java/com/chronos/mobile/core/model/Course.kt
@file:OptIn(kotlinx.serialization.InternalSerializationApi::class)

package com.chronos.mobile.core.model

import androidx.compose.runtime.Immutable
import kotlinx.serialization.Serializable

@Serializable
@Immutable
data class Course(
    val id: String,
    val name: String,
    val teacher: String,
    val location: String,
    val dayOfWeek: Int,
    val startPeriod: Int,
    val endPeriod: Int,
    val color: String,
    val textColor: String = "#21005D",
    val weeks: List<Int> = emptyList(),
)

File: src/lib/models/course.ts
export const COURSE_REMARK_MAX_LENGTH = 200;

export interface Course {
	id: string;
	name: string;
	teacher: string;
	location: string;
	dayOfWeek: number;
	startPeriod: number;
	endPeriod: number;
	color: string;
	textColor: string;
	weeks: number[];
	remark: string;
}

export function createCourse(
	partial: Omit<Course, 'textColor' | 'weeks' | 'remark'> &
		Partial<Pick<Course, 'textColor' | 'weeks' | 'remark'>>
): Course {
	return {
		textColor: '#21005D',
		weeks: [],
		remark: '',
		...partial
	};
}
